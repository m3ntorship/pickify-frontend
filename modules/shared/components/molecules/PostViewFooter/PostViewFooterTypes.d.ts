declare namespace PostViewFooterTypes {
  export interface IProps {
    handleShareClick: () => void;
    voters?: number;
    votersPlaceHolder: string;
    id: string;
  }
}

export { PostViewFooterTypes };
