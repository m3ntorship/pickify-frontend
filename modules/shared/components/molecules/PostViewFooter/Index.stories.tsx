import type { ReactElement } from 'react';
import type { Story } from '@storybook/react';
import PostViewFooter from './Index';
import type { PostViewFooterTypes } from './PostViewFooterTypes';

export default {
  title: 'molecules/PostViewFooter',
  component: PostViewFooter,
};

const Template: Story<PostViewFooterTypes.IProps> = (args): ReactElement => (
  <PostViewFooter {...args} />
);
export const postViewFooter = Template.bind({});
postViewFooter.args = {
  voters: 35432,
  votersPlaceHolder: 'Vote to uncover the total number of voters',
};
