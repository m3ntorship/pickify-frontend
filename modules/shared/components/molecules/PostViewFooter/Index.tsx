import React from 'react';
import type { ReactElement, FC } from 'react';
import type { PostViewFooterTypes } from './PostViewFooterTypes';
import ShareIcon from '../../icons/share.svg';

const PostViewFooter: FC<PostViewFooterTypes.IProps> = ({
  handleShareClick,
  voters,
  votersPlaceHolder,
  id,
}): ReactElement => {
  return (
    <div className="flex justify-between items-center text-grey mt-m">
      <span>{voters ? `${voters} Votes` : votersPlaceHolder}</span>
      <ShareIcon
        onClick={handleShareClick}
        className="fill-grey"
        data-testid={id}
      />
    </div>
  );
};

export default PostViewFooter;
