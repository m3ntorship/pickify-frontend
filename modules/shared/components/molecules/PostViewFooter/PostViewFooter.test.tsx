import * as React from 'react';
import * as renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import PostViewFooter from './Index';
import type { TargetElement } from '@testing-library/user-event';

describe('Testing PostViewFooter component with snapshot', () => {
  it('Testing PostViewFooter without the {voters} should have the votersPlaceHolder value ', () => {
    const tree = renderer
      .create(
        <PostViewFooter
          handleShareClick={(): boolean => {
            return true;
          }}
          votersPlaceHolder="test"
          id="test"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('Testing PostViewFooter with the voters={100} should have voters number ', () => {
    const tree = renderer
      .create(
        <PostViewFooter
          handleShareClick={(): boolean => {
            return true;
          }}
          voters={100}
          votersPlaceHolder="test"
          id="test"
        />,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
describe('Misc', () => {
  it('should render Error Misc', () => {
    const mokingFunction = jest.fn();
    render(
      <PostViewFooter
        handleShareClick={mokingFunction}
        id="test"
        votersPlaceHolder="test"
      />,
    );
    const shareBtn: TargetElement = screen.getByTestId('test');
    userEvent.click(shareBtn);
    expect(mokingFunction).toBeCalled();
  });
});
