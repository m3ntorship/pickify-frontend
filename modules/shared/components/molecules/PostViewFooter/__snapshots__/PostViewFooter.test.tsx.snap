// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing PostViewFooter component with snapshot Testing PostViewFooter with the voters={100} should have voters number  1`] = `
<div
  className="flex justify-between items-center text-grey mt-m"
>
  <span
    className="text-sm"
  >
    100 Votes
  </span>
  <svg
    className="fill-grey"
    data-file-name="SvgShare"
    data-testid="test"
    onClick={[Function]}
  />
</div>
`;

exports[`Testing PostViewFooter component with snapshot Testing PostViewFooter without the {voters} should have the votersPlaceHolder value  1`] = `
<div
  className="flex justify-between items-center text-grey mt-m"
>
  <span
    className="text-sm"
  >
    test
  </span>
  <svg
    className="fill-grey"
    data-file-name="SvgShare"
    data-testid="test"
    onClick={[Function]}
  />
</div>
`;
